modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/capability/xml/schema/builder/ObjectTypeSchemaDelegate.java

```diff
@@ -410,9 +413,27 @@ private TopLevelElement registerAbstractElement(QName typeQName, DslElementSynta
     return abstractElement;
   }
 
+  private Optional<QName> resolveSubstitutionGroupFromString(String userConfiguredSubstitutionGroup) {
+    String[] splittedSubstitutionGroup = userConfiguredSubstitutionGroup.split(":");
+    if (splittedSubstitutionGroup.length == 2) {
+      String namespacePrefix = splittedSubstitutionGroup[0];
+      String substitutionComponent = splittedSubstitutionGroup[1];
+      String namespaceUri = builder.getNamespaceUri(namespacePrefix);
+      if (namespaceUri != null) {
+        return Optional.of(new QName(namespaceUri, substitutionComponent, namespacePrefix));
+      }
+    }
+    return Optional.empty();
+  }
+
   private QName getAbstractElementSubstitutionGroup(DslElementSyntax typeDsl, Optional<DslElementSyntax> baseDsl) {
     QName substitutionGroup;
-    if (baseDsl.isPresent()) {
+    //First check if the substitutionGroup was defined by the user
+    if (!typeDsl.getSubstitutionGroup().isEmpty()) {
+      substitutionGroup = resolveSubstitutionGroupFromString(typeDsl.getSubstitutionGroup())
+          .orElseThrow(() -> new IllegalArgumentException(typeDsl.getSubstitutionGroup()
```

Comment:
resolveSubstitutionGroupFromString() should fail instead of leaving it up to the consumer to do so